lib/trino-orc/src/test/java/io/trino/orc/AbstractTestOrcReader.java

```diff
@@ -507,6 +509,27 @@ public void testEmptyBinarySequence()
         tester.testRoundTrip(VARBINARY, nCopies(30_000, new SqlVarbinary(new byte[0])));
     }
 
+    @Test
+    public void testUUIDDirectSequence()
+            throws Exception
+    {
+        tester.testRoundTrip(
+                UUID,
+                intsBetween(0, 30_000).stream()
+                        .map(i -> randomUUID())
+                        .collect(toList()));
+    }
+
+    @Test
+    public void testUUIDDictionarySequence()
+            throws Exception
+    {
+        tester.testRoundTrip(
+                UUID, ImmutableList.copyOf(limit(cycle(ImmutableList.of(1, 3, 5, 7, 11, 13, 17)), 30_000)).stream()
+                        .map(i -> randomUUID())
```

Comment:
This doesn't work since you get a random UUID each time. The point of this test is to dictionary encode repeated values, so you need to reuse the values for the same integer. Maybe do
```java
Map<Integer, UUID> values = new HashMap<>();
...
.map(i -> values.computeIfAbsent(i, x -> randomUUID()))